src/main/java/org/jetbrains/research/refactorinsight/processors/CommitMiner.java

```diff
@@ -107,19 +163,31 @@ public void consume(GitCommit gitCommit) throws ProcessCanceledException {
           return;
         }
         GitHistoryRefactoringMiner miner = new GitHistoryRefactoringMinerImpl();
+        ExecutorService service = Executors.newSingleThreadExecutor();
+        Future<?> f = null;
         try {
-          miner.detectAtCommit(myRepository,
-              commitId, new RefactoringHandler() {
-                @Override
-                public void handle(String commitId, List<Refactoring> refactorings) {
-                  map.put(commitId,
-                      RefactoringEntry
-                          .convert(refactorings, gitCommit, myProject));
-                  incrementProgress();
-                }
-              });
+          Runnable r = () -> miner.detectAtCommit(myRepository, commitId, new RefactoringHandler() {
```

Comment:
Why we do not use [this method](https://github.com/tsantalis/RefactoringMiner/blob/39bc66358ae08d2719eec8ae0b31a5dac76253a9/src/org/refactoringminer/api/GitHistoryRefactoringMiner.java#L78)?

Reply:
Since we wanted to label the commits that have a timeout, and the method above just returns an empty array of refactorings. This can be helpful in the future if we want to specify which commits take longer than our limit to be mined.